provider/src/main/java/com/synopsys/integration/alert/provider/blackduck/processor/message/service/BomComponent404Handler.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * provider
+ *
+ * Copyright (c) 2021 Synopsys, Inc.
+ *
+ * Use subject to the terms and conditions of the Synopsys End User Software License and Maintenance Agreement. All rights reserved worldwide.
+ */
+package com.synopsys.integration.alert.provider.blackduck.processor.message.service;
+
+import org.jetbrains.annotations.Nullable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+import com.synopsys.integration.rest.exception.IntegrationRestException;
+
+@Component
+public class BomComponent404Handler {
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+
+    public void logIf404OrThrow(IntegrationRestException restException, String componentName, @Nullable String componentVersionName) throws IntegrationRestException {
```

Comment:
This may be more useful as a static method as we won't need to inject this in to classes and no constructor is currently necessary

Reply:
Yeah, it's 50/50, but much easier to add components to the constructor if need be rather than updating every static usage IMO.